smac/utils/io/traj_logging.py

```diff
@@ -172,6 +176,81 @@ def _add_in_aclib_format(self, train_perf: float, incumbent_id: int,
             json.dump(traj_entry, fp)
             fp.write("\n")
 
+    def _add_in_alljson_format(self, train_perf: float, incumbent_id: int,
+                               incumbent: Configuration, ta_time_used: float,
+                               wallclock_time: float):
+        """Adds entries to AClib2-like (but with configs as json) trajectory file
+
+        Parameters
+        ----------
+        train_perf: float
+            Estimated performance on training (sub)set
+        incumbent_id: int
+            Id of incumbent
+        incumbent: Configuration()
+            Current incumbent configuration
+        ta_time_used: float
+            CPU time used by the target algorithm
+        wallclock_time: float
+            Wallclock time used so far
+        """
+        # Silent support for passing configuration-dict's directly:
+        inc_dict = incumbent
+        if isinstance(incumbent, Configuration):
+            inc_dict = incumbent.get_dictionary()
+
+        traj_entry = {"cpu_time": ta_time_used,
+                      "total_cpu_time": None,  # TODO: fix this
+                      "wallclock_time": wallclock_time,
+                      "evaluations": self.stats.ta_runs,
+                      "cost": train_perf,
+                      "incumbent": inc_dict,
+                      }
+        # Silent support for passing configuration-dict's directly (since actual Configuration's always have
+        #   origin-attribute since v0.4.11 and minimum required by SMAC is 0.4.6):
```

Comment:
maybe it would be better to increase the minimum required version

Reply:
Sorry for the unclarity: minimum version is not the problem, we do not need this for backwards-compatibility. I simply assumed, there is a reason for this to support passing dict-objects instead of Configurations (as it is also implemented this way in the tests). If there is not, I will remove silent support and rewrite the tests.

Reply:
At least I cannot remember a reason for why we need to support dicts there. Maybe this is an artifact from the time when we have not used the ConfigSpace.

Reply:
Dropping it then.